packages/plugin-ext/src/main/browser/view/tree-view-widget.tsx

```diff
@@ -384,22 +385,40 @@ export class TreeViewWidget extends TreeViewWelcomeWidget {
 
     handleEnter(event: KeyboardEvent): void {
         super.handleEnter(event);
-        this.tryExecuteCommand();
+        this.tryExecuteCommand(this.findCommands());
     }
 
     handleClickEvent(node: TreeNode, event: React.MouseEvent<HTMLElement>): void {
         super.handleClickEvent(node, event);
-        this.tryExecuteCommand(node);
+        // If clicked on item (not collapsable icon) - execute command or toggle expansion if item has no command
+        if (!event.currentTarget.hasAttribute('data-node-id')) {
+            const commandMap = this.findCommands(node);
+            if (commandMap.size > 0) {
+                this.tryExecuteCommand(commandMap);
+            } else {
+                if (this.isExpandable(node) && !this.hasShiftMask(event) && !this.hasCtrlCmdMask(event)) {
```

Comment:
Suggestion: Please merge this `else` and `if` branch into one `else if`.

Reply:
Thamks, fixed!